controllers/k8ssandra/datacenters.go

```diff
@@ -86,7 +86,7 @@ func (r *K8ssandraClusterReconciler) reconcileDatacenters(ctx context.Context, k
 				return result.Error(err), actualDcs
 			}
 
-			if !utils.CompareAnnotations(actualDc, desiredDc, api.ResourceHashAnnotation) {
+			if !utils.CompareHashes(actualDc, desiredDc) {
```

Comment:
Since we have `AddHashAnnotation` what do you think about `CompareHashAnnotations` for a bit more consistency and clarity?